src/services/completions.ts

```diff
@@ -2054,11 +2070,13 @@ function getEntryForMemberCompletion(
 
 function getPresentModifiers(
     contextToken: Node | undefined,
+    previousToken: Node | undefined,

```

Comment:
you could avoid having to pass `previousToken` here like this: https://github.com/Andarist/TypeScript/commit/4721fbdc61bb64a1f896fc5010b826dcf1b48c50

note that this commit reverts changes to `getPresentModifiers` and introduces changes (slightly adjusted) from https://github.com/microsoft/TypeScript/pull/57480 . I think that your version might have some further readability improvements but I just skipped merging your changes into this. If you like the core of the proposed change - I can incorporate your changes into this and open a PR targeting your branch

Reply:
Does that work because the case where `previousToken` is not the same as `contextToken` is the same as when `contextToken.parent.name !== contextToken`?

Reply:
I believe so, yes.